wandb/sdk/internal/system/assets/gpu_apple.py

```diff
@@ -37,6 +42,25 @@ class _Stats(TypedDict):
     # cpuWaitMs: float
 
 
+def get_apple_gpu_path() -> pathlib.Path:
+    return (
+        pathlib.Path(sys.modules["wandb"].__path__[0]) / "bin" / "apple_gpu_stats"
+    ).resolve()
+
+
+def sample_apple_gpu(executalbe: str) -> dict:
+    try:
+        command = [executalbe, "--json"]
+        output = (
+            subprocess.check_output(command, universal_newlines=True)
+            .strip()
+            .split("\n")
+        )[0]
+        return json.loads(output)
+    except (OSError, ValueError, TypeError, subprocess.CalledProcessError) as e:
```

Comment:
It's not immediately clear what can cause some of these errors, in particular `ValueError` and `TypeError`, so it's also not clear whether all possible errors are caught. Since the intention here is to wrap exceptions in a custom exception type to add more context, you can just use `except Exception as e`.

Reply:
yeah, i think i'll just put it to be like it was before, and we will just fix the swift code to return actual values and have it packaged as go library (probably through cgo)